Sources/NIOCore/AsyncSequences/NIOThrowingAsyncSequenceProducer.swift

```diff
@@ -388,43 +388,64 @@ extension NIOThrowingAsyncSequenceProducer {
 extension NIOThrowingAsyncSequenceProducer {
     /// This is the underlying storage of the sequence. The goal of this is to synchronize the access to all state.
     @usableFromInline
-    /* fileprivate */ internal final class Storage: @unchecked Sendable {
-        /// The lock that protects our state.
+    /* fileprivate */ internal struct Storage: Sendable {
         @usableFromInline
-        /* private */ internal let _lock = NIOLock()
-        /// The state machine.
-        @usableFromInline
-        /* private */ internal var _stateMachine: StateMachine
-        /// The delegate.
+        struct State: Sendable {
+            @usableFromInline
+            var stateMachine: StateMachine
+            @usableFromInline
+            var delegate: Delegate?
+            @usableFromInline
+            var didSuspend: (@Sendable () -> Void)?
```

Comment:
This value was never mutated, so I think we can make it `let` and remove it from the lock.